File: src/main/java/hotels/business/worker/service/WorkerService.java
package hotels.business.worker.service;

import hotels.business.worker.domain.Worker;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.http.ResponseEntity;
import org.springframework.web.multipart.MultipartFile;

/**
 * Created by kutsaniuk on 11.03.16.
 */
public interface WorkerService {

    ResponseEntity<Worker> add( final Worker worker );

    ResponseEntity<Worker> edit( Worker worker );

    ResponseEntity<Void> delete( Long id );

    Page<Worker> search( Pageable pageable, String fullName, String post, String date );

    ResponseEntity<Void> updateLogo( Long id, MultipartFile logo );

    ResponseEntity<String> getLogo( Long id );

}

File: src/main/webapp/app/modules/worker/worker.service.js
(function () {
    'use strict';

    angular.module('main')
    .service('WorkerService', function ($http) {

        var urlBase = '/worker';

        this.getAll = function () {
            return $http.get(urlBase, { 
                    params: { 
                        page: 0, 
                        size: 1000
                    }
                });
        };

        this.get = function (id) {
            return $http.get(urlBase + '/' + id);
        };

        this.new = function (worker) {
            return $http.post(urlBase, worker);
        };

        this.update = function (worker) {
            return $http.put(urlBase, worker)
        };

        this.delete = function (id) {
            return $http.delete(urlBase, { 
                    params: { 
                        id: id
                    }
                }); 
        };

        this.getPage = function (currentPage, size, fullName, post, date) {
            return $http.get(urlBase, { 
                    params: { 
                        page: currentPage, 
                        size: size,
                        fullName: fullName,
                        post: post,
                        date: date
                    }
            });
        };

        this.getImages = function (id) {
            return $http.get(urlBase + '/images', { 
                    params: { 
                        id: id
                    }
            });
        }

        this.deleteImageById = function (id) {
            return $http.delete(urlBase + '/images', { 
                    params: { 
                        id: id
                    }
            });
        }

    });

})();
